main/main.cpp

```diff
@@ -2021,6 +2052,30 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 		}
 	}
 
+#ifdef TOOLS_ENABLED
+	// If the editor is running in windowed mode, ensure the window rect fits
+	// the screen in case screen count or position has changed
+	if (editor && init_windowed) {
+		// We still need to check we are actually in windowed mode, because
+		// certain platform might only support one fullscreen window.
+		if (DisplayServer::get_singleton()->window_get_mode() == DisplayServer::WINDOW_MODE_WINDOWED) {
+			Vector2i current_size = DisplayServer::get_singleton()->window_get_size();
+			Vector2i current_pos = DisplayServer::get_singleton()->window_get_position();
+			int screen = DisplayServer::get_singleton()->window_get_current_screen();
+			Rect2i screen_rect = DisplayServer::get_singleton()->screen_get_usable_rect(screen);
+
+			Vector2i adjusted_end = screen_rect.get_end().min(current_pos + current_size);
+			Vector2i adjusted_pos = screen_rect.get_position().max(adjusted_end - current_size);
+			Vector2i adjusted_size = DisplayServer::get_singleton()->window_get_min_size().max(adjusted_end - adjusted_pos);
+
+			if (current_pos != adjusted_end || current_size != adjusted_size) {
+				DisplayServer::get_singleton()->window_set_position(adjusted_pos);
+				DisplayServer::get_singleton()->window_set_size(adjusted_size);
+			}
+		}
+	}
+#endif
```

Comment:
It might be better to do it in the `DisplayServer` constructors instead, since doing it after can case already shown window to jump, or its size can be unnecessarily adjusted by OS/WM.

Relevant code:

https://github.com/godotengine/godot/blob/27253f3eb2c78a9ad5114c92eae2036b10e1d7e0/platform/windows/display_server_windows.cpp#L4209-L4217
https://github.com/godotengine/godot/blob/27253f3eb2c78a9ad5114c92eae2036b10e1d7e0/platform/linuxbsd/x11/display_server_x11.cpp#L5641-L5649
https://github.com/godotengine/godot/blob/27253f3eb2c78a9ad5114c92eae2036b10e1d7e0/platform/macos/display_server_macos.mm#L3878-L3886